app/backend/server/routers/dates_router.py

```diff
@@ -0,0 +1,19 @@
+from fastapi import APIRouter, Query, Depends, HTTPException
+from typing import List
+from datetime import date, time
+from utils.datetime_utils import daterange
+
+from db.models import *
+from db.database import transaction
+from db.crud import get_multiple_users_redused_dates_data
+from db import schemas
+
+router = APIRouter()
+
+
+@router.get("/api/v1/dates_status", response_model=schemas.MultipleUsersDatesResponce)
+async def get_dates_status(start: date, end: date = None, users_id: List[int] = Query([])):
+    return get_multiple_users_redused_dates_data(db=Depends(transaction),
```

Comment:
Depends (transaction) might not work with contextmanager- look it up

Reply:
move it to params so the function will be more testable later on

Reply:
move what to params? didnt understand

Reply:
move db to a param for the function
get_dates_status(db = Depends(transaction))

Reply:
done, tho i dont know how it will be seen in the swagger